### PR TITLE
MAINT: Remove uses of scalar aliases

### PR DESCRIPTION
**Pull request by eric-wieser**
Relates to gh-6103

Proof that this is exhaustive as of writing can be found in the CI of gh-14882, hopefully.

Even if it's not exhaustive, it's still good to merge something that gets us closer.

Split from gh-14882 so that cleanup can go in quickly, assuming a long discussion about deprecation.

### CODE REVIEW

File: numpy/random/_bounded_integers.pyx.in [hunk 1]
```diff
@@ -191,7 +191,7 @@ cdef object _rand_{{nptype}}_broadcast(object low, object high, object size,
         highm1_arr = <np.ndarray>np.PyArray_FROM_OTF(high_m1, np.{{npctype}}, np.NPY_ALIGNED | np.NPY_FORCECAST)
     else:
         # If input is object or a floating type
-        highm1_arr = <np.ndarray>np.empty_like(high_arr, dtype=np.{{nptype}})
+        highm1_arr = <np.ndarray>np.empty_like(high_arr, dtype=np.{{otype}})
```
Comment (eric-wieser) [inline numpy/random/_bounded_integers.pyx.in]: A note: nptype is such that `npy_{nptype}` is valid in C, otype is such that `np.{otype}` is valid in python.

File: numpy/tests/test_public_api.py [hunk 2]
```diff
@@ -23,7 +23,10 @@ def check_dir(module, module_name=None):
         module_name = module.__name__
     results = {}
     for name in dir(module):
-        item = getattr(module, name)
+        # record but discard warnings for deprecated attributes
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter('always')
+            item = getattr(module, name)
```
Comment (eric-wieser) [inline numpy/tests/test_public_api.py]: Oops, this doesn't belong in this commit

File: numpy/core/tests/test_numeric.py [hunk 3]
```diff
@@ -2953,7 +2953,7 @@ def test_sparse(self):
         assert_array_equal(x, np.array([[0], [1], [2], [3]]))
         assert_array_equal(y, np.array([[0, 1, 2]]))
 
-    @pytest.mark.parametrize("dtype", [np.int, np.float32, np.float64])
+    @pytest.mark.parametrize("dtype", [int, np.float32, np.float64])
```
Comment (mattip) [inline numpy/core/tests/test_numeric.py]: maybe `np.int64` to parallel the other dtypes? It seems a bit odd to use a builtin here

File: numpy/core/tests/test_ufunc.py [hunk 4]
```diff
@@ -980,7 +980,7 @@ def test_can_ignore_signature(self):
         assert_array_equal(out, mm_row_col_vec.squeeze())
 
     def test_matrix_multiply(self):
-        self.compare_matrix_multiply_results(np.long)
+        self.compare_matrix_multiply_results(np.compat.long)
```
Comment (mattip) [inline numpy/core/tests/test_ufunc.py]: maybe a more specific dtype would be better than using a builtin?

**Comment by mattip** [comment]
LGTM. The test failure for b7181a3 is the win32 heisenbug.  There are a few places where the use of a builtin for test parametrization next to specific dtypes looks out of place. Maybe they could all be specific dtypes rather than a mix of specific and builtins, but other than that this is a nice cleanup.

File: numpy/core/tests/test_ufunc.py (hunk 4 above)
Comment (eric-wieser) [reply numpy/core/tests/test_ufunc.py]: How about I make a follow-up that uses parametrize for this test?

**Comment by eric-wieser** [comment]
A few of these tests really are testing the behavior of builtins. I fixed the two you called out.

File: numpy/core/tests/test_ufunc.py (hunk 4 above)
Comment (mattip) [reply numpy/core/tests/test_ufunc.py]: ok

**Comment by mattip** [comment]
Thanks @eric-wieser
